riscv-pages/src/sequential_pages.rs

```diff
@@ -236,6 +264,43 @@ impl<S: State, I: Iterator<Item = Page<S>>> fmt::Debug for Error<S, I> {
     }
 }
 
+pub struct SeqPageChunkIter<S: State> {
+    pages: SequentialPages<S>,
+    n: u64,
+}
+
+impl<S: State> Iterator for SeqPageChunkIter<S> {
+    type Item = SequentialPages<S>;
+
+    fn next(&mut self) -> Option<SequentialPages<S>> {
```

Comment:
This code looks very similar to `take` above, can the iterator re-use that code?

Reply:
Agreed. I was in fact meant to do that but clearly forgot.